File: CountUpAnimation/app/src/main/java/com/example/jestina/myapplication/utils/YWLog.java
package com.example.jestina.myapplication.utils;

import android.util.Log;

/**
 * Created by JESTINA on 2018-08-24.
 *
 */
public class YWLog {
    private static final String TAG = "====>";

    public static void d(String debugMsg) {
        String logStr = buildLogMsg(debugMsg);
        Log.d(TAG, logStr);
    }

    private static String buildLogMsg(String message) {
        StackTraceElement ste = Thread.currentThread().getStackTrace()[4];
        StringBuilder sb = new StringBuilder();
        sb.append("[");
        sb.append(ste.getFileName());
        /*
         * sb.append(" > "); sb.append(ste.getMethodName());
         */
        sb.append(" > #");
        sb.append(ste.getLineNumber());
        sb.append("] ");
        sb.append(message);
        return sb.toString();
    }
}

File: CountUpAni_Kt/app/src/main/java/com/example/jestina/countupani_kt/utils/YWLog.kt
package com.example.jestina.countupani_kt.utils

import android.util.Log

/**
 * Created by JESTINA on 2018-09-01.
 */
class YWLog {
    companion object {
        val TAG = "====>"

        fun d(debugMsg: String) {
            val logStr = buildLogMsg(debugMsg)
            Log.d(TAG, logStr)
        }


        private fun buildLogMsg(message: String): String {
            val ste = Thread.currentThread().stackTrace[4]
            val sb = StringBuilder()
            sb.append("[")
            sb.append(ste.fileName)
            /*
             * sb.append(" > "); sb.append(ste.getMethodName());
             */
            sb.append(" > #")
            sb.append(ste.lineNumber)
            sb.append("] ")
            sb.append(message)
            return sb.toString()
        }
    }
}